drivers/SmartThings/matter-switch/src/sub_drivers/camera/init.lua

```diff
@@ -0,0 +1,1558 @@
+  if #optional_capabilities ~= #prev_optional_capabilities then
+    return true
+  end
+  for _, capability in pairs(optional_capabilities) do
+    if not switch_utils.tbl_contains(prev_optional_capabilities, capability) then
+      return true
+    end
+  end
+  for _, capability in pairs(prev_optional_capabilities) do
+    if not switch_utils.tbl_contains(optional_capabilities, capability) then
+      return true
+    end
+  end
+  return false
+end
+
+local function add_camera_subscribed_attributes(device)
+  if #device:get_endpoints(clusters.CameraAvStreamManagement.ID) > 0 then
+    device:add_subscribed_attribute(clusters.CameraAvStreamManagement.attributes.AttributeList)
+  end
+  for capability, attr_list in pairs(subscribed_attributes) do
+    if device:supports_capability_by_id(capability) then
+      for _, attr in pairs(attr_list) do
+        device:add_subscribed_attribute(attr)
+      end
+    end
+  end
+  for capability, event_list in pairs(subscribed_events) do
+    if device:supports_capability_by_id(capability) then
+      for _, event in pairs(event_list) do
+        device:add_subscribed_event(event)
+      end
+    end
+  end
+  if device:get_field(IS_PARENT_CHILD_DEVICE) then
+    local device_type_attribute_map = {
+      [ON_OFF_LIGHT_DEVICE_TYPE_ID] = {
+        clusters.OnOff.attributes.OnOff
+      },
+      [DIMMABLE_LIGHT_DEVICE_TYPE_ID] = {
+        clusters.OnOff.attributes.OnOff,
+        clusters.LevelControl.attributes.CurrentLevel,
+        clusters.LevelControl.attributes.MaxLevel,
+        clusters.LevelControl.attributes.MinLevel
+      },
+      [COLOR_TEMP_LIGHT_DEVICE_TYPE_ID] = {
+        clusters.OnOff.attributes.OnOff,
+        clusters.LevelControl.attributes.CurrentLevel,
+        clusters.LevelControl.attributes.MaxLevel,
+        clusters.LevelControl.attributes.MinLevel,
+        clusters.ColorControl.attributes.ColorTemperatureMireds,
+        clusters.ColorControl.attributes.ColorTempPhysicalMaxMireds,
+        clusters.ColorControl.attributes.ColorTempPhysicalMinMireds
+      },
+      [EXTENDED_COLOR_LIGHT_DEVICE_TYPE_ID] = {
+        clusters.OnOff.attributes.OnOff,
+        clusters.LevelControl.attributes.CurrentLevel,
+        clusters.LevelControl.attributes.MaxLevel,
+        clusters.LevelControl.attributes.MinLevel,
+        clusters.ColorControl.attributes.ColorTemperatureMireds,
+        clusters.ColorControl.attributes.ColorTempPhysicalMaxMireds,
+        clusters.ColorControl.attributes.ColorTempPhysicalMinMireds,
+        clusters.ColorControl.attributes.CurrentHue,
+        clusters.ColorControl.attributes.CurrentSaturation,
+        clusters.ColorControl.attributes.CurrentX,
+        clusters.ColorControl.attributes.CurrentY
+      }
+    }
+    for _, ep in ipairs(device.endpoints) do
+      local id = 0
+      for _, dt in ipairs(ep.device_types) do
+        id = math.max(id, dt.device_type_id)
+      end
+      for _, attr in pairs(device_type_attribute_map[id] or {}) do
```

Comment:
```suggestion
      for _, attr in pairs(fields.device_type_attribute_map[id] or {}) do
```
let's remove the re-defined table and just use the one in fields

Reply:
we can also drop the re-defining of the device type fields here